projects/plugins/super-cache/inc/boost.php

```diff
@@ -11,6 +11,60 @@ function wpsc_get_boost_migration_config() {
 	);
 }
 
+/**
+ * Display an admin notice to install Jetpack Boost.
+ */
+function wpsc_jetpack_boost_notice() {
+	// Don't show the banner if the banner has been dismissed.
+	$is_dismissed = '1' === get_user_option( 'wpsc_dismissed_boost_admin_notice' );
+	if ( $is_dismissed ) {
+		return;
+	}
+
+	// Don't show the admin notice if Jetpack Boost is too old.
+	if ( ! wpsc_is_boost_current() ) {
+		return;
+	}
+
+	// Don't show the banner if Super Cache is using features that Boost doesn't support.
+	if ( ! wpsc_is_boost_compatible() ) {
+		return;
+	}
+
+	$config       = wpsc_get_boost_migration_config();
+	$button_url   = $config['is_installed'] ? $config['activate_url'] : $config['install_url'];
+	$button_class = $config['is_installed'] ? 'wpsc-activate-boost-button' : 'wpsc-install-boost-button';
+
+	?>
+	<div id="wpsc-notice-boost-migrate" class="notice boost-notice notice-success is-dismissible">
+	<h3>
+		<?php esc_html_e( 'Migrate to Jetpack Boost', 'wp-super-cache' ); ?>
+	</h3>
+	<p>
+		<?php esc_html_e( 'Your WP Super Cache setup is compatible with Boost\'s new caching feature. Continue to cache as you currently do and enhance your site\'s speed using our highly-rated performance solutions.', 'wp-super-cache' ); ?>
+	</p>
+
+	<p>
+		<div class="wpsc-boost-migration-error" style="display:none; color:red; margin-bottom: 20px;"></div>
+		<a data-source='notice' class='button button-primary <?php echo esc_attr( $button_class ); ?>' href="<?php echo esc_url( $button_url ); ?>">
+			<div class="spinner" style="display:none; margin-top: 8px"></div>
+			<label><?php esc_html_e( 'Migrate now', 'wp-super-cache' ); ?></label>
+		</a>
+	</p>
+	</div>
+	<?php
+}
+if ( isset( $_GET['page'] ) && $_GET['page'] === 'wpsupercache' ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	add_action( 'admin_notices', 'wpsc_jetpack_boost_notice' );
+}
```

Comment:
I might be missing context about where is this file included, but shouldn't this be the other way around - I mean the `$_GET` check inside `wpsc_jetpack_boost_notice`?

Reply:
It's loaded by wp-cache.php, so it's loaded by the frontend and backend, but I've done it this way before, to stop loading the function unnecessarily when the stop condition was simple. Your argument for doing the other way would be to keep the function self contained?

Reply:
Mostly that. It just seems like a micro-optimization and I'm not sure I see the value in doing that. I've always leaned towards keeping hooks self-contained.

Reply:
Fair enough. I moved the checks inside. Makes sense.